### PR TITLE
AndroidMediaCodec: Retry VC1 fix

### PR DESCRIPTION
**Pull request by dbarkar**
## Description
VC1 mime type should be replaced only when appropriate file is opened. 
Currently, Kodi returns VC1 codec every time it enumerates it, if required codec comes further, the video can't be opened.
Only Sony TV (or any other device that supports `video/VC1` or `video/vc1` types) are affected.

See https://github.com/xbmc/xbmc/pull/24488 and https://github.com/xbmc/xbmc/pull/24491.

May be related to https://github.com/xbmc/xbmc/issues/24496.

## Motivation and context
Easy fix for the issue until there is a better mechanism.
The replacement code is properly isolated now.

## How has this been tested?
Tested locally on Sony TV.

## What is the effect on users?
Fixes playback for many file types on Sony TV (and maybe other devices).

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by fritsch** [comment]
Thanks!

**Review by fritsch — APPROVED** [review]

**Comment by dbarkar** [comment]
> I'm not sure if logic is correct: I mean now when is not vc1 always is executed continue; and maybe is skipped other codec ¿?
> 
> not should be:
> 
> ```
> if (m_mime == "video/wvc1")
> {
>   do extra vc1 stuff
>   . . . . 
>   if (!success)
>     continue;
> }
> ```
> 
> ?

Sorry, I don't see how proper codec can be skipped, can you please elaborate or give an example? On continue new codec is tested.

This code should only be executed when it's vc1 codec and vc1 mime type, both conditions must be true,.

**Comment by dbarkar** [comment]
Oh, I see...
Thanks, looks better now.

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp [hunk 1]
```diff
@@ -751,39 +751,42 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
...
+          {
+            m_mime = v;
+            CLog::Log(LOGDEBUG, "Succesfully replaced VC1 mime type to {}", m_mime);
+            success = true;
+            break;
+          }
         }
+        if (!success)
+          continue;
       }
```
Comment (thexai) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:789]: after close `if (m_mime == "video/wvc1")`  is missing

```c++
else
{
  continue;
}
```

Comment (dbarkar) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp:789]: good catch, thanks!

**Review by thexai — APPROVED** [review]
Changes looks better hiding white space:
https://github.com/xbmc/xbmc/pull/24516/files?diff=unified&w=1

**Review by fritsch — APPROVED** [review]

**Comment by fritsch** [comment]
jenkins build this please

**Pull request merged**